### ISSUE TITLE
Creating datatables the angular way causes re-creating/applying of datable elements on existing table

### ISSUE THREAD

**Issue by afaqi** [l-lin/angular-datatables#171]
The problem occurs when I have a table built using ng-repeat and it also includes some ng-if on it. Also to mention that I am using bootstrap-ui accordion which includes this table within. All those accordions are built using ng-repeat and so the table rows

	<accordion close-others="false"> 
               <accordion-group ng-repeat="<some-logic>">
			<accordion-heading>
			</accordion-heading>
                                    <table class="table" id="<someId>" style="width:100%" datatable="ng" ng-if="item.open">
                   </accordion-group>
![accordion-problem-1st](https://cloud.githubusercontent.com/assets/240198/5676707/2f77dcac-97e3-11e4-8a33-975487c919b5.png)
![accordion-problem-2nd](https://cloud.githubusercontent.com/assets/240198/5676705/2f7274b0-97e3-11e4-8c8b-f8add0d66c6d.png)

       </accordion>

Now, this behavior has worked perfectly fine for me before using datable. After adding datatables, I see that it is adding the header element of datatable (which includes number of items per page and search field) on everytime accordion is reopened.

In short, when I open accordion first time, it works ok and datable shown as expected. I close and re-open then it creates another search-header at bottom of existing table and keep adding if I keep repeating the process. Unfortunately, I cannot share the code since it is private to company and it is not trivial to create a functioning example for it on plukr. I highly appreciate your time if you could look into it. As of my understanding ng-if removes and recreates a clone of the element but somehow it seems it tries to reapply some datatable style on it. Please check images.

Also to mention, this problem does not occur if I remove ng-if on the table but then it breaks certain things in my code because angular does not accept it therefore I need to keep ng-if. Thanks a lot in advance.

**Comment by afaqi** [comment]
As you can see. First image is how it should be. Second one is after I closed the accordion and re-opened it again. If header does not apply in bottom, it should be perfectly fine. Everytime I repeat, it creates one more.

**Comment by l-lin** [comment]
I can't find any difference between these two images. Can you point out where the issue is in the picture?

Have you tried to use `ng-show` instead of `ng-if`? `ng-if` removes or recreates the portion of the DOM whereas `ng-show` shows or hides the given HTML element.

**Comment by afaqi** [comment]
The open accordion in second image has also "show 10 entries" (which seems to be header of datatables). It is duplicated on everytime this accordion is closed and reopened. Also to mention, the datatable reload event is fired as well.

**Comment by afaqi** [comment]
ng-show does not make any difference in this case. I believe it could be a  problem with logic of datatables rather than angular-datatables.

**Comment by l-lin** [comment]
If it's something to toggle, then it's better to use `ng-show`. It will render only once and the `DataTable` will be called only once (thus the request sent to the server is only sent once).

I think you can solve your issue by putting your `ng-show` in a level above of your table.
See [plnkr](http://embed.plnkr.co/fiu8jVsBcCTgFaIQsoIh/) for example.
Notice that `ng-show` is "faster".

**Comment by afaqi** [comment]
lin! Thank you very much for your time and very useful help. Yes, I used your solution and it seems to be working fine at least. ng-show works fine for parent node of table (not well on table itself). I will check it further though. :), thanks again!
